docs: add map_delete function (#898)
datafuselabs/databend-docs: docs/en/sql-reference/20-sql-functions/10-map-functions/map-delete.md

---
title: MAP_DELETE
---
import FunctionDescription from '@site/src/components/FunctionDescription';

<FunctionDescription description="Introduced or updated: v1.2.547"/>

> Returns a MAP based on an existing MAP with one or more keys removed.

Reviewer: ```suggestion
Returns an existing MAP with one or more keys removed.
```

## Syntax

```sql
MAP_DELETE( <map>, <key1> [, <key2>, ... ] )
```

## Arguments

| Arguments | Description                                  |
|-----------|----------------------------------------------|
| `<map>`   | The MAP that contains the KEY to remove.     |
| `<keyN>`  | The KEY to be omitted from the returned MAP. |

:::note

> - The type of the key expression must match the type of the map’s key.

Reviewer: ```suggestion
- The types of the key expressions and the keys in the map must be the same.
```

> - Key values that are not found in the map are ignored.

Reviewer: ```suggestion
- Key values not found in the map will be ignored.
```
